spec/datadog/core/configuration/config_helper_spec.rb

```diff
@@ -0,0 +1,407 @@
+      let(:env_vars) { { 'DD_UNSUPPORTED_VAR' => 'value' } }
+      let(:supported_configurations) { {} }
+
+      # For now, datadog-ci-rb is not supported and we don't want to break it.
+      it 'returns the environment variable value even if unsupported' do
+        is_expected.to eq('value')
+      end
+    end
+
+    context 'when environment variable starts with DD_ but is not supported' do
+      let(:name) { 'DD_UNSUPPORTED_VAR' }
+      let(:env_vars) { { 'DD_UNSUPPORTED_VAR' => 'value' } }
+      let(:supported_configurations) { {} } # Override to make it unsupported
+
+      context 'when not in test environment' do
+        around do |example|
+          described_class.instance_variable_set(:@raise_on_unknown_env_var, nil)
+
+          example.run
+
+          described_class.instance_variable_set(:@raise_on_unknown_env_var, true)
+        end
+
+        it 'returns nil' do
+          is_expected.to be_nil
+        end
+      end
+
+      context 'when in test environment' do
+        it 'raises an error for unsupported DD_ variables' do
+          expect { subject }.to raise_error(RuntimeError, /Missing DD_UNSUPPORTED_VAR env\/configuration/)
+        end
+      end
+    end
+
+    context 'when using a deprecated alias that does not start with DD_ or OTEL_' do
+      let(:name) { 'DISABLE_DATADOG_RAILS' }
+      let(:env_vars) { { 'DISABLE_DATADOG_RAILS' => 'true' } }
+      let(:supported_configurations) { {'DD_DISABLE_DATADOG_RAILS' => { version: ['A'] } } }
+      let(:aliases) { { 'DISABLE_DATADOG_RAILS' => ['DD_DISABLE_DATADOG_RAILS'] } }
+      let(:alias_to_canonical) { { 'DISABLE_DATADOG_RAILS' => 'DD_DISABLE_DATADOG_RAILS' } }
+
+      context 'when not in test environment' do
+        around do |example|
+          described_class.instance_variable_set(:@raise_on_unknown_env_var, nil)
+
+          example.run
+
+          described_class.instance_variable_set(:@raise_on_unknown_env_var, true)
+        end
+
+        it 'returns the environment variable value' do
+          is_expected.to eq(nil)
+        end
+      end
+
+      context 'when in test environment' do
+        it 'raises an error suggesting the canonical name' do
+          expect { subject }.to raise_error(RuntimeError, /Please use DD_DISABLE_DATADOG_RAILS instead/)
+        end
+      end
+    end
+
+    context 'when environment variable does not start with DD_ or OTEL_' do
+      let(:name) { 'SOME_OTHER_VAR' }
+      let(:env_vars) { { 'SOME_OTHER_VAR' => 'value' } }
+
+      it 'returns the environment variable value' do
+        is_expected.to eq('value')
+      end
+    end
+  end
+
+  describe '.log_deprecated_environment_variables' do
+    let(:mock_io) { StringIO.new }
+    let(:mock_logger) { Datadog::Core::Logger.new(mock_io) }
+
+    before do
+      # Reset instance variables that might be set by previous tests
+      described_class.instance_variable_set(:@log_deprecations_called_with, nil)
+    end
```

Comment:
Similar note as for `@raise_on_unknown_env_var`: If this becomes part of the `ConfigHelper` instance, then it doesn't need to be reset as every instance will have its own copy -- but the instance set in `DATADOG_ENV` is always the same so we still get the behavior we want.